src/control/JacobianController.h

```diff
@@ -0,0 +1,139 @@
+#pragma once
+
+#include "../filters/StateSpaceUtil.h"
+#include "../navtypes.h"
+#include "../world_interface/data.h"
+#include "TrapezoidalVelocityProfile.h"
+
+#include <array>
+#include <functional>
+#include <optional>
+
+#include <Eigen/Core>
+
+namespace control {
+
+/**
+ * @brief This class controls the position of a multidimensional mechanism
+ * with a trapezoidal velocity profile using the jacobian of the kinematics.
+ *
+ * @tparam outputDim The dimension of the output vector of the kinematic function.
+ * @tparam inputDim The dimension of the input vector of the kinematic function.
+ * @see https://en.wikipedia.org/wiki/Jacobian_matrix_and_determinant
+ */
+template <int outputDim, int inputDim> class JacobianController {
```

Comment:
Not sure how in depth you want to go with the math, but I think it would be helpful to explicitly state that the dimensions of the jacobian function is: kinematicOutputDim x kinematicInputDim in the description.